timeseries/src/autogluon/timeseries/models/multi_window/multi_window_model.py

```diff
@@ -0,0 +1,190 @@
+            self.model_base: AbstractTimeSeriesModel = model_base
+        self.model_base_type = type(self.model_base)
+        self.info_per_val_window = []
+
+        self.most_recent_model: AbstractTimeSeriesModel = None
+        self.most_recent_model_path: str = None
+        super().__init__(**kwargs)
+
+    def _fit(
+        self,
+        train_data: TimeSeriesDataFrame,
+        val_data: Optional[TimeSeriesDataFrame] = None,
+        time_limit: Optional[int] = None,
+        num_val_windows: int = 1,
+        **kwargs,
+    ):
+        # TODO: use incremental training for GluonTS models?
+        # TODO: implement parallel fitting similar to ParallelLocalFoldFittingStrategy in tabular?
+        verbosity = kwargs.get("verbosity", 2)
+        set_logger_verbosity(verbosity, logger=logger)
+
+        if val_data is not None:
+            raise ValueError(f"val_data should not be passed to {self.name}.fit()")
+        if num_val_windows == 0:
+            raise ValueError("MultiWindowBacktestingModel can only be trained with num_val_windows > 0")
+
+        trained_models = []
+        global_fit_start_time = time.time()
+        for window_idx in range(num_val_windows):
+            train_fold, val_fold = train_data.train_test_split(
+                prediction_length=self.prediction_length,
+                window_idx=window_idx,
+                suffix=f"_W{window_idx}",
+            )
+
+            logger.debug(f"\tWindow {window_idx}")
+            model = self.get_child_model(window_idx)
+            model_fit_start_time = time.time()
+            model.fit(
+                train_data=train_fold,
+                val_data=val_fold,
+                time_limit=None if time_limit is None else time_limit - (model_fit_start_time - global_fit_start_time),
+                **kwargs,
+            )
+            model.fit_time = time.time() - model_fit_start_time
+            model.score_and_cache_oof(val_fold, store_val_score=True, store_predict_time=True)
+            trained_models.append(model)
+
+            logger.debug(f"\t\t{model.val_score:<7.4f}".ljust(15) + f"= Validation score ({model.eval_metric})")
+            logger.debug(f"\t\t{model.fit_time:<7.3f} s".ljust(15) + "= Training runtime")
+            logger.debug(f"\t\t{model.predict_time:<7.3f} s".ljust(15) + "= Training runtime")
+
+            self.info_per_val_window.append(
+                {
+                    "window_idx": window_idx,
+                    "fit_time": model.fit_time,
+                    "val_score": model.val_score,
+                    "predict_time": model.predict_time,
+                }
+            )
+
+        # Only the model trained on most recent data is saved & used for prediction
+        self.most_recent_model = trained_models[0]
+        self.most_recent_model_path = trained_models[0].path
+        self.predict_time = self.most_recent_model.predict_time
+        self.fit_time = time.time() - global_fit_start_time - self.predict_time
```

Comment:
Since in Tabular prediction is done with an ensemble consisting of all trained models, there `fit_time = sum(model.fit_time for model in bagged_models)` and `predict_time = sum(model.predict_time for model in bagged_models)`.

My idea here was to satisfy the following conditions:
- time that we spent fitting & scoring the model (excluding metric computation time) = `fit_time + predict_time` 
- time that it actually takes to predict with `MultiWindowBacktestingModel` = `predict_time`

Does this sound reasonable?

Reply:
"fit_time = sum(model.fit_time for model in bagged_models)" Not actually the case in Tabular. This used to be the logic, but after we added parallel bagging this is no longer the logic, instead it is the end-to-end wall-clock time of the fit call of the bagged ensemble, although you can check the code for the nuance.

Reply:
Thanks, Nick! Here we have sequential fitting of folds, so I think what we report is equal to both the end-to-end wall-clock time and the sum of fit_times for individual models. I agree that if we implement parallel fold fitting here, we should report the wall-clock time, like in ParallelFoldFittingStrategy in Tabular.